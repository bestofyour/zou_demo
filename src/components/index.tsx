import * as React from 'react';
interface Name {
  name?: string;
}
import { add } from 'ramda';

export class ShoppingList extends React.Component<Name, Record<string, unknown>> {
  render() {
    return (
      <div className="shopping-list">
        <h1>Shopping List for {this.props.name}</h1>
        <ul>
          <li>Instagram</li>
          <li>WhatsApp</li>
          <li>Oculus</li>
        </ul>
      </div>
    );
  }
}

interface Props {
  value?: number;
  age?: number;
}

class Square extends React.Component<Props & Name, Record<string, unknown>> {
  render() {
    return <button className="square">{this.props.value}</button>;
  }
}

export class Board extends React.Component {
  renderSquare(i: number) {
    return <Square value={add(i, 100)} />;
  }

  render() {
    const status = 'Next player: X';

    return (
      <div>
        <div className="status">{status}</div>
        <div className="board-row">
          {this.renderSquare(0)}
          {this.renderSquare(1)}
          {this.renderSquare(2)}
        </div>
        <div className="board-row">
          {this.renderSquare(3)}
          {this.renderSquare(4)}
          {this.renderSquare(5)}
        </div>
        <div className="board-row">
          {this.renderSquare(6)}
          {this.renderSquare(7)}
          {this.renderSquare(8)}
        </div>
      </div>
    );
  }
}
