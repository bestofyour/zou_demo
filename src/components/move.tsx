import * as React from 'react';

type Props = Record<string, unknown>;

export const MoveBox = (props: Props): React.ReactElement => {
  console.log(props);
  return <div className="box-out move"></div>;
};
