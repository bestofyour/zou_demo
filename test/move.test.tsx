import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MoveBox } from '../src/components/move';

// describe 描述一组测试
describe('test Movebox component render', () => {
  it('test Movebox component render success', () => {
    // renderer.create 模拟 React 渲染组件，并将结果输出为 json 字符串。
    const moveComp = renderer.create(<MoveBox></MoveBox>).toJSON();
    // 期望 moveComp 匹配生成的快照文件。
    expect(moveComp).toMatchSnapshot();
  });
});