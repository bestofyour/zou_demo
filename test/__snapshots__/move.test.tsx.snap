// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Movebox component render test Movebox component render success 1`] = `
<div
  className="box-out move"
/>
`;
